src/input.cpp

```diff
@@ -98,246 +96,146 @@ create_rapi_json() {
-              break;
-            case 7:
-              estate = "No Earth Ground";
-              break;
-            case 8:
-              estate = "Stuck Relay";
-              break;
-            case 9:
-              estate = "GFCI Self Test Failed";
-              break;
-            case 10:
-              estate = "Over Temperature";
-              break;
-            case 254:
-              estate = "Sleeping";
-              break;
-            case 255:
-              estate = "Disabled";
-              break;
-            default:
-              estate = "Invalid";
-              break;
-           }
-         }
-       }
-     }
-    if (rapi_command_sent == 0 && rapi_command == 3) {
-       Serial.println("$GG*B2");
-      }
-    if (rapi_command_sent == 1 && rapi_command == 3) {
-      while (Serial.available()) {
-         String rapiString = Serial.readStringUntil('\r');
-        if (rapiString.startsWith("$OK")) {
-           comm_success++;
-          int firstRapiCmd = rapiString.indexOf(' ');
-          int secondRapiCmd = rapiString.indexOf(' ', firstRapiCmd + 1);
-          amp = rapiString.substring(firstRapiCmd + 1, secondRapiCmd);
-          volt = rapiString.substring(secondRapiCmd + 1);
-         }
-       }
-     }
-    if (rapi_command_sent == 0 && rapi_command == 4) {
-       Serial.println("$GP*BB");
-     }
-    if (rapi_command_sent == 1 && rapi_command == 4) {
-      while (Serial.available()) {
-         String rapiString = Serial.readStringUntil('\r');
-        if (rapiString.startsWith("$OK")) {
-          comm_success++;
-          int firstRapiCmd = rapiString.indexOf(' ');
-          int secondRapiCmd = rapiString.indexOf(' ', firstRapiCmd + 1);
-          int thirdRapiCmd = rapiString.indexOf(' ', secondRapiCmd + 1);
-          temp1 = rapiString.substring(firstRapiCmd + 1, secondRapiCmd);
-          temp2 = rapiString.substring(secondRapiCmd + 1, thirdRapiCmd);
-          temp3 = rapiString.substring(thirdRapiCmd + 1);
-         }
-      }
-    }
-    if (rapi_command_sent == 0 && rapi_command == 5) {
-      Serial.println("$GU*C0");
-    }
-    if (rapi_command_sent == 1 && rapi_command == 5) {
-      while (Serial.available()) {
-      String rapiString = Serial.readStringUntil('\r');
-        if (rapiString.startsWith("$OK")) {
-        comm_success++;
-        int firstRapiCmd = rapiString.indexOf(' ');
-          int secondRapiCmd = rapiString.indexOf(' ', firstRapiCmd + 1);
-          wattsec = rapiString.substring(firstRapiCmd + 1, secondRapiCmd);
-          watthour_total = rapiString.substring(secondRapiCmd + 1);
+  comm_sent++;
+  if (rapi_command == 1) {
+    if (0 == rapiSender.sendCmd("$GE"))
+      comm_success++;
+      pilot = rapiSender.getToken(1);
+  }
+  if (rapi_command == 2) {
+    if (0 == rapiSender.sendCmd("$GS")) {
+      comm_success++;
+      String qrapi = rapiSender.getToken(1);
+      state = strtol(qrapi.c_str(), NULL, 16);
+      switch (state) {
+        case 1:
```

Comment:
Can this be an #define or enum, 'magic' numbers are bad for code readability (same comment for every number here)